Lib/glyphsLib/builder/builders.py

```diff
@@ -387,17 +387,37 @@ def _valid_designspace(self, designspace):
         """Make sure that the user-provided designspace has loaded fonts and
         that names are the same as those from the UFOs.
         """
-        # TODO: (jany) really make a copy to avoid modifying the original object
-        copy = designspace
-        for source in copy.sources:
+        import copy
+
+        designspace_copy = copy.copy(designspace)
+
+        sources = []
+        sources_to_rename = []
+        for source in designspace_copy.sources:
```

Comment:
if you want to avoid modifying the input designspace object (while still avoiding calling a fallible deepcopy over defcon.Fonts), you could make a shallow copy of the SourceDescriptors here

```suggestion
        for source in designspace_copy.sources:
            source_copy = copy.copy(source)
```